src/compiler/utilities.ts

```diff
@@ -1401,11 +1401,10 @@ namespace ts {
 
     /// Given a BinaryExpression, returns SpecialPropertyAssignmentKind for the various kinds of property
     /// assignments we treat as special in the binder
-    export function getSpecialPropertyAssignmentKind(expression: ts.BinaryExpression): SpecialPropertyAssignmentKind {
-        if (!isInJavaScriptFile(expression)) {
+    export function getSpecialPropertyAssignmentKind(expr: ts.BinaryExpression): SpecialPropertyAssignmentKind {

```

Comment:
Why was the parameter renamed?

Reply:
In the old version it was immediately assigned to a `const expr` -- this cuts out the middleman.
You could of course rename the declaration to `expression` if you want, along with all of its uses.